heron/spi/src/java/com/twitter/heron/spi/statefulstorage/IStatefulStorage.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.spi.statefulstorage;
+
+import java.util.Map;
+
+public interface IStatefulStorage {
+  /**
+   * Initialize the Storage Backend
+   *
+   * @param conf An unmodifiableMap containing basic configuration
+   * Attempts to modify the returned map,
+   * whether direct or via its collection views, result in an UnsupportedOperationException.
+   */
+  void init(Map<String, Object> conf);
+
+  /**
+   * Closes the Checkpoints Backend
+   */
+  void close();
+
+  // Store the checkpoint
+  boolean store(Checkpoint checkpoint);
```

Comment:
Can we make these return `void` and throw one or more custom exceptions upon failure, instead of using the "booleans to represent failures" anti-pattern? See https://github.com/twitter/heron/issues/1311